### PR TITLE
[Feature, Documentation, DEV-147] basic framework in place

### PR DESCRIPTION
**Pull request by davesag**
* added basic framework for API server and database connection with roles and users
* trying out setting up postgres on circleci

### CODE REVIEW

File: test/unit/utils/db/dbUrlParser_spec.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+const parseDatabaseUrl = require('../../../../src/utils/db/dbUrlParser')
+
+describe('parseDatabaseUrl', () => {
+  const dbUrl =
+    'postgres://pfforbjhkrletg:aic5oO6Cran1g3hk6mJa5QqNZB@ec2-23-21-91-97.compute-1.amazonaws.com:5432/dek11b2j1g3mfb'
+  const expected = {
+    database: 'dek11b2j1g3mfb',
+    username: 'pfforbjhkrletg',
+    password: 'aic5oO6Cran1g3hk6mJa5QqNZB',
```
Comment (tavakyan) [inline test/unit/utils/db/dbUrlParser_spec.js:11]: this doesn't seem secure

**Review by tavakyan — CHANGES_REQUESTED** [review]
Everything looks fine, but could we use a .env file?

File: test/unit/utils/db/dbUrlParser_spec.js (hunk 1 above)
Comment (davesag) [reply test/unit/utils/db/dbUrlParser_spec.js:11]: it's fake test data, not real data.  It's just testing that the url format heroku uses can be parsed correctly.

**Comment by davesag** [comment]
environment variables will override what's in `config.json`.  the `config.json` is there to simplify development, that's all.  IN production Heroku will sent a `DATABASE_URL` environment variable which is parsed in preference to what's in the `config.json` file.

**Pull request merged**